README.md

```diff
@@ -8,13 +8,202 @@
+
+Like I mentioned, this `color` is now an Object *property* and it corresponds to
+the value `'orange'`. We also refer to Object properties as *keys*, the combination of the
+key and value colloquially referred to as a *key-value pair*.
+
+Let's add the rest of our data. Note that we can use any variable type when
+assigning keys / properties to Objects.
+
+```javascript
+var cat = {};
+cat.color = 'orange'; // string
+cat.pattern = 'striped';
+cat.declawed = true; // boolean
+cat.weight = 10; // number
+```
+
+We can also use the shorthand Object constructor to write the equivalent:
+
+```javascript
+var cat = {
+  color: 'orange',
+  pattern: 'striped',
+  declawed: true,
+  weight: 10
+};
+```
+
+Or, alternatively, just use quotation marks:
+
+```javascript
+var cat = {};
+cat['color'] = 'orange'; // string
+cat['pattern'] = 'striped';
+cat['declawed'] = true; // boolean
+cat['weight'] = 10; // number
+```
+
+These methods are all identical, and the keys as strings example (above) is
+for when you have oddly named properties.
+
+## Modifying a Key-Value pair
+
+Once you've created your object, the properties (or keys) attached to your
+object are *mutable*, which means to say they can change. (Think of the word
+  mutation. You're mutating the value.) They can also be removed completely.
+
+If we take our example above, and we do:
+
+```
+console.log(cat.color);
+```
+
+We'll see the output
+
+```
+'orange'
+```
+
+We can change this with
+
+```
+cat.color = 'brown';
+```
+
+We can also remove the property altogether with:
+
+```
+delete cat.color;
+```
+
+Now,
+
+```
+console.log(cat.color) // => undefined
+```
+
+And if we log the cat Object, we get:
+
+```
+{
+  pattern: 'striped',
+  declawed: true,
+  weight: 10
+}
+```
+
+No color property to be found!
+
+## Iterating over Key-Value pairs
+
+To loop over every key-value pair in an Object, you can use a `for in` loop.
+
+```
+for (key in cat) {
+  var value = cat[key];
+  console.log(key, value);
+}
+```
+
+**Please note**: The behavior of this loop is *not* alphabetical, and you should
+not rely on the specific order of keys for any code behavior. Also note that this
+loop outputs the *keys* of the Object, and you must retrieve the value manually.
+
+## Advanced Iteration
+
+Don't worry, this is a little bit advanced, but fun to look at now because
+you'll likely use it a lot down the road. If you'd like to iterate over the keys
+in alphabetical order, here's a snippet:
+
+```
+var keys = Object.keys(cat); // Retrieve the keys as an Array
+key.sort(); // sort the Array (alphabetically)
+
+var i, key, value;
+for (i = 0; i < keys.length; i++) {
+  key = keys[i];
+  value = cat[key];
+  console.log(key, value);
+}
+```
+
+This can be further reduced (using array methods...):
+
+```
+Object.keys(cat).sort().forEach(function(key) {
+  var value = cat[key];
+  console.log(key, value);
+});
+```
+
+Where `Object.keys(cat)` is an array, `.sort()` sorts the array and returns it,
```

Comment:
This is a great example to include, but students are going to feel a bit overwhelmed -- `Object.keys()` is totally new, so it'd maybe be good to spend some time explaining what's going on there and walking students through an example or two.